index.ts

```diff
@@ -259,20 +259,60 @@ async function splitNeuron(neuronId: bigint, amount: bigint) {
   ok();
 }
 
-async function spawnNeuron(neuronId: string, controller?: Principal) {
+async function spawnNeuron(neuronId: string, controller?: Principal, percentage?: number) {
   const identity = await getLedgerIdentity();
+  // Percentage is only supported with Candid transaction which was added in version 2.2.0
+  if (percentage !== undefined) {
+    await assertLedgerVersion({ identity, minVersion: "2.2.0" });
+  }
   const governance = GovernanceCanister.create({
     agent: await getAgent(identity),
-    hardwareWallet: true,
+    // `hardwareWallet: true` uses Protobuf and doesn't support percentage
+    hardwareWallet: await isCurrentVersionSmallerThan({ identity, version: "0.2.2" })
+      && percentage === undefined,
   });
 
   const spawnedNeuronId = await governance.spawnNeuron({
     neuronId: BigInt(neuronId),
     newController: controller,
+    percentageToSpawn: percentage,
   });
   ok(`Spawned neuron with ID ${spawnedNeuronId}`);
 }
 
+async function stakeMaturity(neuronId: bigint, percerntage?: number) {
+  const identity = await getLedgerIdentity();
+  await assertLedgerVersion({ identity, minVersion: "2.2.0" });
+  const governance = GovernanceCanister.create({
+    agent: await getAgent(identity),
+    hardwareWallet: false,
```

Comment:
Sorry, I don't know the background, why `hardwareWallet: false`?

Reply:
we use this to define when we want to use the protobuf transactions instead of the candid ones.

I will remove then, the default is `false` and it's confusing to set it to so inside a harware wallet cli.

Reply:
Thanks for the explanation.